MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+        // Only add the region if it is not already there
+        if (!region) {
+            NSEntityDescription *newRegionDescription = [NSEntityDescription entityForName:@"Region" inManagedObjectContext:self.managedObjectContext];
+            region = [[Region alloc] initWithEntity:newRegionDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            region.name = regionDict[@"region_name"];
+            region.id = regionDict[@"_id"];
+            region.keyName = regionDict[@"keyName"];
+        }
+        // regardless if the region is new or not, we should populate it with area data
+        // But first, we need to check that the region file exists
+        NSArray *areaDrops = [self loadArrayFromJsonFileNamed:[NSString stringWithFormat:@"%@_drops", region.keyName]];
+        // If the file actually exists...
+        if (areaDrops) {
+            // ... go through the file, and construct areas/drops as appropriate
+            for (NSDictionary *areaDrop in areaDrops) {
+                //This is getting complicated, so I'm going to move this out to a separate method.
+                [self addAreaDropFromDictionary:areaDrop toRegion:region];
+            }
+        }
+    }
+    
+    // We should now try to save the manged object context
+    if (![self didSuccessfullySaveContext]) {
+        NSLog(@"WARNING: Failed to save the item context!");
+    }
+}
+
+- (void)addAreaDropFromDictionary:(NSDictionary *)areaDropDict toRegion:(Region *)region
+{
+    //First, let's check if the area that this areaDrop takes place in actually exists!
+    NSString *areaCombinedName = [NSString stringWithFormat:@"%@-%@-%@",
+                                  region.name,
+                                  areaDropDict[@"area"],
+                                  areaDropDict[@"rank"]];
+    Area *area = [self getUniqueAreaWithCombinedName:areaCombinedName];
+    // If the area doesn't already exist...
+    if (!area) {
+        // ... We need to create it, and add it to this region
+        NSEntityDescription *newAreaDescription = [NSEntityDescription entityForName:@"Area" inManagedObjectContext:self.managedObjectContext];
+        area = [[Area alloc] initWithEntity:newAreaDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        area.name = areaDropDict[@"area"];
+        area.combinedName = areaCombinedName;
+        area.region = region;
+        area.rank = areaDropDict[@"rank"];
+        [region.areaSet addObject:area];
+    }
+    // If it wasn't already, the area is now attached to the region
+    
+    // We should now consider adding this drop to the area.
+    // First, check if the area drop already exists, by way of checking the idDecimalString
+    NSString *idDecimalString = areaDropDict[@"_id"];
+    AreaDrop *drop = [self getUniqueAreaDropWithDecimalID:idDecimalString];
+    //If the drop already exists...
+    if (drop) {
+        // ... do nothing. (At least for now).
```

Comment:
If you put in a comment with `TODO: ` (the `:` is necessary, I'm not sure if the space after the `:` is necessary), Xcode will usually pick up on it and show it in the drop-down navigator.  It's also convenient to be able to search for a known tag like `TODO` to find all your things to come back to later.